openlibrary/plugins/admin/code.py

```diff
@@ -311,16 +311,28 @@ def POST_block_account_and_revert(self, account):
         i = 0
         edits = 0
         stop = False
+        keys_to_delete = set()
         while not stop:
             changes = account.get_recentchanges(limit=100, offset=100 * i)
+            added_records: list[list[dict]] = [
+                c.changes for c in changes if c.kind == 'add-book'
+            ]
+            flattened_records: list[dict] = sum(added_records, [])
+            keys_to_delete |= {r['key'] for r in flattened_records}
```

Comment:
If we want to be extra careful, we can add the key to `keys_to_delete` only if `r['revision'] == 1` here.

Reply:
This seems right -- I think we only want to delete either their changes or the document if they created it. Otherwise, this all looks like it's the right direction.